src/extension.ts

```diff
@@ -93,7 +92,12 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
             if (!node) {
                 node = await ext.tree.showTreeItemPicker(accountContextValues.map((val: string) => val += AttachedAccountSuffix), actionContext);
             }
-
+            if (node instanceof MongoAccountTreeItem) {
+                const nodes = await node.getCachedChildren(actionContext);
```

Comment:
Rather than use `getCachedChildren`, check out `ext.connectedMongoDB` [here](https://github.com/microsoft/vscode-cosmosdb/blob/master/src/extensionVariables.ts). You should be able to check it's parent to determine if you need to disconnect